sync/class.jetpack-sync-listener.php

```diff
@@ -301,4 +302,8 @@ function set_defaults() {
 		$this->set_queue_size_limit( Jetpack_Sync_Settings::get_setting( 'max_queue_size' ) );
 		$this->set_queue_lag_limit( Jetpack_Sync_Settings::get_setting( 'max_queue_lag' ) );
 	}
+
+	function get_request_url() {
+		return 'http' . (isset($_SERVER['HTTPS']) ? 's' : '') . '://' . "{$_SERVER['HTTP_HOST']}{$_SERVER['REQUEST_URI']}";
+	}
```

Comment:
rather relying on $_SERVER here, which can be unreliable, I recommend using, the approach outlined here:

https://konstantin.blog/2012/current-url-in-wordpress/